### PR TITLE
Adds support to build promql parser in CI

### PR DESCRIPTION
**Pull request by Harkishen-Singh**
Signed-off-by: Harkishen-Singh <harkishensingh@hotmail.com>

Fixes: #7488 

From pipeline
```
>> running check for promql-parser build
go get -u golang.org/x/tools/cmd/goyacc
go: found golang.org/x/tools/cmd/goyacc in golang.org/x/tools v0.0.0-20200702044944-0cc1aa72b347
goyacc -o promql/parser/generated_parser_temp.y.go promql/parser/generated_parser.y
conflicts: 2 shift/reduce
diff -I '^// Code generated by goyacc -o.*' -I '^//line.*' promql/parser/generated_parser_temp.y.go promql/parser/generated_parser.y.go
rm promql/parser/generated_parser_temp.y.go y.output
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -62,8 +61,16 @@ react-app-test: | $(REACT_APP_NODE_MODULES_PATH) react-app-lint
 	@echo ">> running React app tests"
 	cd $(REACT_APP_PATH) && yarn test --no-watch --coverage
 
+.PHONY: promql-parser
+promql-parser:
+	@echo ">> running check for promql-parser build"
+	go get -u golang.org/x/tools/cmd/goyacc
```
Comment (roidelapluie) [inline Makefile:67]: somehow that would change the go mod?

Comment (roidelapluie) [reply Makefile:67]: I think goyacc should be installed in circleci config, not in the makefile

**Comment by Harkishen-Singh** [comment]
@roidelapluie can you give a look at why is the test failing, and a possible suggestion to fix it?

**Comment by Harkishen-Singh** [comment]
@roidelapluie any views?

**Comment by Harkishen-Singh** [comment]
ping @roidelapluie

**Review by codesome — COMMENTED** [review]
LGTM in general, but I have less context around this one. @roidelapluie could you give it another look since you had some comments?

**Pull request closed without merging**